lighthouse-core/lib/dependency-graph/simulator/network-analyzer.js

```diff
@@ -319,6 +320,17 @@ class NetworkAnalyzer {
     const estimatesByOrigin = NetworkAnalyzer._estimateResponseTimeByOrigin(records, rttByOrigin);
     return NetworkAnalyzer.summarize(estimatesByOrigin);
   }
+
+  /**
+   * @param {Array<LH.WebInspector.NetworkRequest>} records
+   * @return {LH.WebInspector.NetworkRequest}
+   */
+  static findMainDocument(records) {
+    // TODO(phulce): handle more edge cases like client redirects, or plumb through finalUrl
```

Comment:
adding `finalUrl` to all calls of request* is just insane work atm, should be easier after a refactor to pass in artifacts and make that dance a little easier